file_test.go

```diff
@@ -1056,3 +1057,105 @@ func TestFile__RemoveBatch(t *testing.T) {
 		t.Errorf("unexpected number of return entries: %d", len(file.ReturnEntries))
 	}
 }
+
+func TestFile__SegmentFile(t *testing.T) {
+	// open a file for reading. Any io.Reader Can be used
+	f, err := os.Open(filepath.Join("examples", "ach-ppd-read-mixedDebitCredit", "ppd-mixedDebitCredit.ach"))
+	// f, err := os.Open(filepath.Join("test", "ach-ppd-credit", "ppd-credit.ach"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	r := NewReader(f)
+	achFile, err := r.Read()
+	if err != nil {
+		t.Fatalf("Issue reading file: %+v \n", err)
+	}
+
+	// ensure we have a validated file structure
+	if achFile.Validate(); err != nil {
+		t.Fatalf("Could not validate entire read file: %v", err)
+	}
+
+	creditFile, debitFile, err := achFile.SegmentFile()
```

Comment:
Should we validate the number of batches in each file and some details?

Reply:
Do you want to validate against the original File?

Reply:
We could probably do that in the service_test as well.  Simialr to

// TestGetBatches return a list of batches for the supplied file.id
func TestGetBatches(t *testing.T) {
	s := mockServiceInMemory()
	batches := s.GetBatches("98765")
	if len(batches) != 1 {
		t.Errorf("expected %s received %v", "1", len(batches))
	}
}

Either I guess is fine.